### PR TITLE
added brave product urls to brave://about

### PR DESCRIPTION
**Pull request by nullhook**
Resolves https://github.com/brave/brave-browser/issues/17299

image

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Comment by bsclifton** [comment]
Not sure if it's due to Chromium 93 merging? or if it's Windows specific (unlikely) but this isn't working for me unfortunately

Screencap of what does show up:
image

**Comment by goodov** [comment]
@bsclifton make sure the patch is applied

**Comment by bsclifton** [comment]
@goodov good call! Total rookie mistake there 😂 (didn't put branch name in package.json when running init). OK looks like it's got a failed patch after 93 merged

File: patches/chrome-common-webui_url_constants.cc.patch
Comment (goodov) [inline patches/chrome-common-webui_url_constants.cc.patch:9]: Since the placing of this line is irrelevant for the runtime, I'd suggest putting it right on the first line so we minimize any possible chrome conflicts in the future.
```
const char* const kChromeHostURLs[] = {
    BRAVE_HOST_URLS
    kChromeUIAboutHost,
```

Comment (goodov) [reply patches/chrome-common-webui_url_constants.cc.patch:9]: but actually, we can substitute our hosts by redefining any of ```content::``` hosts and don't introduce the patch at all.

Comment (bsclifton) [reply patches/chrome-common-webui_url_constants.cc.patch:9]: ++ on this; this was the cause of the failed patch 😄

Comment (bsclifton) [reply patches/chrome-common-webui_url_constants.cc.patch]: Hope you don't mind @nullhook - I just rebased this branch and then moved the line to the top 😄 Will give this a go

**Comment by bsclifton** [comment]
Rebased and updated patch 😄

**Review by bsclifton — APPROVED** [review]
++

File: patches/chrome-common-webui_url_constants.cc.patch
Comment (goodov) [reply patches/chrome-common-webui_url_constants.cc.patch]: @bsclifton what about removing the patch?
> we can substitute our hosts by redefining any of ```content::``` hosts and don't introduce the patch at all.
```
#include "chrome/common/webui_url_constants.h"

#define kChromeUIAppCacheInternalsHost \
    kChromeUIAppCacheInternalsHost,
    BRAVE_HOST_URLS

...
```

**Comment by Tonev** [comment]
Since this PR is related to `brave://about`, it might be a good idea to remove/fix `brave://memories` as it leads to nowhere currently.

Reported in Brave Community: https://community.brave.com/t/brave-memories-not-available/273961

File: patches/chrome-common-webui_url_constants.cc.patch
Comment (goodov) [reply patches/chrome-common-webui_url_constants.cc.patch]: @nullhook ^^
> what about removing the patch?

**Comment by bsclifton** [comment]
Thanks for following up on that @Tonev - and thanks in general for being super helpful with issue triage 😄 I definitely appreciate the help you've brought here

There is an issue tracking the removal of that - https://github.com/brave/brave-browser/issues/15582

I'll leave it up to @nullhook if we want to solve in the same pull request 😄 (otherwise, I'm happy to fix)

File: patches/chrome-common-webui_url_constants.cc.patch
Comment (bsclifton) [reply patches/chrome-common-webui_url_constants.cc.patch]: @goodov which page would you suggest we substitute? It should be one that doesn't work / isn't used

If `brave://memories` is in the content:: area, that would be a great way to close out two issues in one go

Comment (goodov) [reply patches/chrome-common-webui_url_constants.cc.patch]: why it should be the one that's not used?
I'm not talking about replacing the content of some variable. I'm talking about using ```#define``` for some a variable that is not defined inside ```webui_url_constants.h```.

Comment (nullhook) [reply patches/chrome-common-webui_url_constants.cc.patch]: such a clever override!

> the fact that we have content::  variables declared in other file allowed this trick. otherwise the patch is the only way

**Comment by nullhook** [comment]
Thanks the pointing the issue @Tonev - let's do this on another PR @bsclifton

File: chromium_src/chrome/common/webui_url_constants.cc
Comment (goodov) [inline chromium_src/chrome/common/webui_url_constants.cc:16]: please ```#undef``` defines after this include:
```
#undef kChromeUIAppCacheInternalsHost
#undef BRAVE_HOST_URLS
```

**Review by goodov — APPROVED** [review]

File: chromium_src/chrome/common/webui_url_constants.cc
Comment (nullhook) [reply chromium_src/chrome/common/webui_url_constants.cc:14]: done

File: patches/chrome-common-webui_url_constants.cc.patch
Comment (bsclifton) [reply patches/chrome-common-webui_url_constants.cc.patch]: Woah - nice! Now I see what you mean. Good call, @goodov 😄

**Review by bsclifton — APPROVED** [review]

**Pull request merged**
